lib/srv/ctx.go

```diff
@@ -722,6 +722,18 @@ func (c *ServerContext) ExecCommand() (*ExecCommand, error) {
 		requestType = c.request.Type
 	}
 
+	var hostname string
+	var err error
+	serverInfo := c.srv.GetInfo()
+	if serverInfo != nil {
+		hostname = serverInfo.GetHostname()
+	} else {
+		hostname, err = os.Hostname()
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
```

Comment:
It doesn't, that's why this is here. Tests provide it with a `FakeServer` that returns nil.

Reply:
You can update `FakeServer` to return a valid value. This code should only be called from a Teleport node, which should always have this value set since it's how it's heartbeat is sent.

https://github.com/gravitational/teleport/blob/master/lib/srv/regular/sshserver.go#L697-L721

Reply:
Updated